src/qibo/models/double_bracket.py

```diff
@@ -1,5 +1,8 @@
+from copy import deepcopy
 from enum import Enum, auto
+from functools import partial
 
+import hyperopt
```

Comment:
Now that we removed the `hyperopt` dep from the main deps, when you try to import this module it will raise a `ModuleNotFoundError`.
I think that you should move this import directly inside the `hyperopt_step` possibly in a try-except block where you can tell the user to install hyperopt if they want to perform the optimiziation.
Perhaps at this point it is easier just to keep it as a dependency.
What do you recommend @scarrazza? Shall we add `hyperopt` to the main deps or we keep it as an optional dep?

Reply:
> I think that you should move this import directly inside the hyperopt_step possibly in a try-except block where you can tell the user to install hyperopt if they want to perform the optimiziation.

I agree with this option